### ISSUE TITLE
[a11y]: Radio buttons not working for JAWS screenreader as expected

### ISSUE THREAD

**Issue by claramannion** [carbon-design-system/carbon#11211]
### Package

carbon-components

### Browser

Edge

### Operating System

Windows

### Package version

10.56.0

### React version

16.13.1

### Automated testing tool and ruleset

manual test

### Assistive technology

JAWS 2020

### Description

Focus goes to the Carbon radio buttons using JAWS. But when user uses UP/DOWN arrow keys it reads Radio button with label and state and when the user uses UP/DOWN arrow key again the focus navigates to the span text and JAWS reads the label. 

This will confuse screenreader users and also not the expected behaviour of JAWS and radio buttons. 

same issue is reproducible on [Carbon-design radio button](https://www.carbondesignsystem.com/components/radio-button/usage#live-demo)

### CodeSandbox example

https://codesandbox.io/s/ecstatic-drake-3kcv4e

### Steps to reproduce

1. Using the latest Edge browser and JAWS 2020
2. Navigate to the radio buttons and then just press the down arrow key
3. ISSUE: JAWS reads out the radio label at the input checkbox and then on next down arrow keypress, it reads out the text aswell. 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by joshblack** [comment]
@dakahn is this related to #11167 at all or is this a different issue?

**Comment by dakahn** [comment]
@joshblack no, i dont believe so

@claramannion JAWS is modular and will enter 'forms mode' automatically for a very speicfic subset of form components (more on that [here in their usage docs](https://www.freedomscientific.com/SurfsUp/Forms.htm)). For all other form elements not contained within forms the user is going to have to do the switching into forms mode themselves. There are two ways to get this done with a radio button
1. when you <kbd>tab</kbd> to a radio button group hit <kb>enter</kb> which should switch to forms mode allowing you to select the radio buttons as expected with the arrow keys
2. when on a page with a radio button group use the <kbd>a</kbd> key to jump between radio buttons and make your selection

As far as i can tell this is working as intended since the above described controls are working for me. Going to close this one up, but feel free to ping me to reopen if i've missed something. 🏄🏾
